### PR TITLE
docs: add icon migration notes

### PR DESCRIPTION
**Pull request by balibebas**
@with-heart Thanks for pointing me to the correct locations to update in #922. I made a fork this time so I can keep this branch up-to-date with `master` so the PR doesn't go stale. Please note this PR leaves off the one place I did change in #922 as it occurred to me migration notes might not be needed in that package (assuming it's new).

I also spotted some more icons feedback in the #906 today. If you'd like me to tackle some of those with this PR to make it more meaningful please don't hestitate to ask.

### CODE REVIEW

**Comment by with-heart** [comment]
IF you'd like to tackle some of the other `Icon` feedback, please do! That would be awesome

**Comment by balibebas** [comment]
Went though the feedback from @beforan and udpated the docs. I think I hit all the icon-related items they had save for using `<Icon as={FaSomeIcon}` as I'm not sure if there are plans to deprecate `Icon`. Possibly contentious change is that I removed the info about **Fallback icons** and I believe those won't be necessary anymore due to the explicit nature of importing icons now.

Please review and merge if it looks good. This is a big stumbling block for the upgrade.

File: website/docs/components/icon.mdx [hunk 1]
```diff
@@ -82,63 +82,26 @@ Here's a working example:
...
-All Chakra icons are stored in the theme object under the `icons` key. You can
-extend this object to add your own icons. Here are the steps:
-
-- Export the icon's `svg` from Figma, Sketch, etc.
-- Use a tool like [SvgOmg](https://svgomg.firebaseapp.com) to reduce the size
-  and minify the markup.
-- Add the icon to the theme object.
-
-> Add the `fill=currentColor` attribute to the `path` or `g` so that when you
-> use `<Icon color="gray.200"/>`, it works correctly.
```
Comment (balibebas) [inline website/docs/components/icon.mdx:94]: Removed as icons are no longer part of theme. I did move the instructions (with fixed SvgOmg link) to another location.

File: website/docs/components/icon.mdx [hunk 2]
```diff
@@ -82,63 +82,26 @@ Here's a working example:
...
-`question-outline` icon.
+- Export the icon's `svg` from Figma, Sketch, etc.
+- Use a tool like [SvgOmg](https://jakearchibald.github.io/svgomg/) to reduce
+  the size and minify the markup.
+- Use `generateIcon` to create an icon as shown above.
 
-```jsx
-<Icon name="naruto" />
-```
+Start using your icon right away. There is no need to update your theme files.
```
Comment (balibebas) [inline website/docs/components/icon.mdx:104]: This is a pretty gnarly diff. Here I'm removing the info about Icon fallbacks as explained in the PR comments. If this needs to come back please let me know. I'm about to go AFK for the night.

File: website/docs/migration.mdx [hunk 3]
```diff
@@ -441,6 +441,46 @@ package which automatically configures `InitializeColorMode` along with
...
+  > Replacement logic: If `<Icon name="search" />` is used, then replace it with
+  > `<SearchIcon />` from Chakra icons package.
+
+```diff
+- import { Icon } from "@chakra-ui/core"
+- <Icon name="search" />
++ import { SearchIcon } from "@chakra-ui/icons"
++ <SearchIcon />
+```
+
```
Comment (balibebas) [inline website/docs/migration.mdx:461]: Here we introduce the Icons package which isn't really discussed anywhere outside Buttons (if at all) and show how to use it to upgrade.

File: website/docs/migration.mdx [hunk 4]
```diff
@@ -441,6 +441,46 @@ package which automatically configures `InitializeColorMode` along with
...
+})
+```
+
+As a convenience you can also import `createIcon` from the `icons` package along
+with other icons:
+
+```jsx live=false
+import { createIcon, MoonIcon, SunIcon } from "@chakra-ui/icons"
+```
+
```
Comment (balibebas) [inline website/docs/migration.mdx:483]: Not sure if this was intentional but it's a very nice affordance and possilby worth documenting.

File: website/docs/theming/theme.mdx [hunk 5]
```diff
@@ -282,12 +282,12 @@ with width set to `1rem` or `16px` and height set to `0.75rem` or `12px`
...
-We recommend trying to solve stacking issues without `z-index`. You can read
-our recommendations and learn more about 
+We recommend trying to solve stacking issues without `z-index`. You can read our
+recommendations and learn more about
 [stacking contexts and z-index here](/z-index).
 
-Chakra provides a minimal set of z-indeces out of the box to help
-control the stacking order of components.
+Chakra provides a minimal set of z-indeces out of the box to help control the
+stacking order of components.
```
Comment (balibebas) [inline website/docs/theming/theme.mdx:290]: Formatting only (automatic due to another change).

File: website/docs/theming/theme.mdx [hunk 6]
```diff
@@ -299,58 +299,6 @@ export default {
...
-// Step 3: Pass the new theme to the ThemeProvider
-
-<ThemeProvider theme={customTheme}>
-  {/* app goes here */}
-</ThemeProvider>
-
-// Step 4: Now you can do this
-<Icon name="icon1" color="green.200"/>
-```
-
```
Comment (balibebas) [inline website/docs/theming/theme.mdx:353]: :wave:

**Comment by balibebas** [comment]
Squashed commits and rebased fork against `upstream/master`.

**Comment by with-heart** [comment]
I think we still have some work to do in `website/docs/components/icon.mdx`. The `Usage` section still refers to `props.name` and `theme.icons`, and doesn't really explain that `Icon`'s purpose is now for creating custom icons. I think we should handle that in a separate PR though so we can at least get these changes in.

**Comment by with-heart** [comment]
Let's continue conversation in #998

**Comment by with-heart** [comment]
@all-contributors please add @balibebas for docs

**Pull request merged**